src/solvers/flattening/boolbv_index.cpp

```diff
@@ -308,28 +244,22 @@ bvt boolbvt::convert_index(
   // the array (constant and variable indexes) and updated
   // versions of it.
 
-  const bvt &tmp=convert_bv(array); // recursive call
-
-  mp_integer offset=index*width;
-
-  if(offset>=0 &&
-     offset+width<=mp_integer(tmp.size()))
+  const bvt &tmp = convert_bv(array);
+  const mp_integer offset = index * width;
+  const bool in_bounds =
+    offset >= 0 && offset + width <= mp_integer(tmp.size());
+  if(in_bounds)
   {
-    // in bounds
-
     // The assertion below is disabled as we want to be able
     // to run CBMC without simplifier.
     // Expression simplification should remove these cases
-    // assert(array.id()!=ID_array_of &&
-    //       array.id()!=ID_array);
```

Comment:
Why remove part of a comment?

Reply:
Sorry I thought it wasn't part of the comments but just code that was commented out. I will rephrase the comment.

Reply:
Actually if the function should work without simplifier why would we need these assertions? I think we should remove the whole comment instead.

Reply:
I agree that the entire comment should be removed.